src/sequence_batch_scheduler.cc

```diff
@@ -109,12 +110,73 @@ SequenceBatchScheduler::Create(
   // runner. This is at least 1 even if the model doesn't support
   // batching.
   const size_t model_batch_size = std::max(1, config.max_batch_size());
-  size_t seq_slot_cnt = model_batch_size;
+  sched->seq_slot_cnt_ = model_batch_size;
   if (config.sequence_batching().has_oldest()) {
-    seq_slot_cnt =
+    sched->seq_slot_cnt_ =
         config.sequence_batching().oldest().max_candidate_sequences();
   }
 
+  // Create batchers.
+  sched->CreateBatchers();
+
+  // Create a reaper thread that watches for idle sequences. Run the
+  // reaper a lower priority.
+  SequenceBatchScheduler* raw = sched.release();
+
+  raw->reaper_thread_exit_ = false;
+  raw->reaper_thread_.reset(
+      new std::thread([raw]() { raw->ReaperThread(10 /* nice */); }));
+
+  scheduler->reset(raw);
+
+  return Status::Success;
+}
+
+Status
+SequenceBatchScheduler::Update(
+    std::unique_ptr<std::lock_guard<std::mutex>>* lock)
+{
+  std::unique_lock<std::mutex> lk(mu_);
+
+  // Instruct 'Enqueue()' to begin pausing new sequence.
+  if (updating_) {
+    return Status(
+        Status::Code::INTERNAL, "sequence batch scheduler is already updating");
+  }
+  updating_ = true;
+
+  // Wait until all in-flight and backlog sequence are queued.
+  update_enqueued_cv_.wait(lk, [this] {
+    return sequence_to_batcherseqslot_map_.empty() && backlog_queues_.empty() &&
+           sequence_to_backlog_map_.empty();
+  });
+
+  // Flush all pending requests to the rate limiter.
+  lk.unlock();
+  batchers_.clear();
```

Comment:
Instead of deleting and creating the batcher objects again during the update, shouldn't we handle these case by case basis?
case 1: Increment in instance count: No need to pause/clear out the request. Just prepare the new batchers for the new instance count.
case 2: Decrement in the instance count: Wait only for the requests in sequences allocated to batchers(for the delta instances) to complete before destructing the instances.

In the current workflow, the scheduler is still doing much of the heavy-lifting with each model instance update.

Reply:
Yes. This is implemented in the latest refresh.